[CI] ListenableActionFutureTests.testListenersNotifiedOnCorrectThreads failed

This looks like a newly added test.

**Build scan**:
https://gradle-enterprise.elastic.co/s/benbyofejft46/tests/:server:test/org.elasticsearch.action.support.ListenableActionFutureTests/testListenersNotifiedOnCorrectThreads#1

**Repro line**:
`./gradlew ':server:test' --tests "org.elasticsearch.action.support.ListenableActionFutureTests.testListenersNotifiedOnCorrectThreads" -Dtests.seed=6F4C360FA7AEA154 -Dtests.security.manager=true -Dtests.locale=und -Dtests.timezone=America/Panama -Druntime.java=11`

**Reproduces locally?**:
Nope.

**Applicable branches**:
Only `master` so far.

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.action.support.ListenableActionFutureTests&tests.disabledDistributions=WyJvdXRjb21lOnNraXBwZWQiXQ&tests.sortField=FAILED&tests.test=testListenersNotifiedOnCorrectThreads&tests.unstableOnly=true

**Failure excerpt**:
```

org.elasticsearch.action.support.ListenableActionFutureTests > testListenersNotifiedOnCorrectThreads FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=336, name=adder-3, state=RUNNABLE, group=TGRP-ListenableActionFutureTests]

        Caused by:
        java.lang.AssertionError
            at __randomizedtesting.SeedInfo.seed([6F4C360FA7AEA154]:0)
            at org.junit.Assert.fail(Assert.java:86)
            at org.junit.Assert.assertTrue(Assert.java:41)
            at org.junit.Assert.assertTrue(Assert.java:52)
            at org.elasticsearch.action.support.ListenableActionFutureTests.lambda$testListenersNotifiedOnCorrectThreads$0(ListenableActionFutureTests.java:105)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=339, name=completer-6, state=RUNNABLE, group=TGRP-ListenableActionFutureTests]

        Caused by:
        java.lang.AssertionError: 
        Expected: "adder-3"
             but: was "completer-6"
            at __randomizedtesting.SeedInfo.seed([6F4C360FA7AEA154]:0)
            at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
            at org.junit.Assert.assertThat(Assert.java:956)
            at org.junit.Assert.assertThat(Assert.java:923)
            at org.elasticsearch.action.support.ListenableActionFutureTests$3.onResponse(ListenableActionFutureTests.java:97)
            at org.elasticsearch.action.support.ListenableActionFutureTests$3.onResponse(ListenableActionFutureTests.java:93)
            at org.elasticsearch.action.support.ListenableActionFuture.executeListener(ListenableActionFuture.java:89)
            at org.elasticsearch.action.support.ListenableActionFuture.done(ListenableActionFuture.java:72)
            at org.elasticsearch.common.util.concurrent.BaseFuture.set(BaseFuture.java:133)
            at org.elasticsearch.action.support.AdapterActionFuture.onResponse(AdapterActionFuture.java:58)
            at org.elasticsearch.action.support.ListenableActionFutureTests.lambda$testListenersNotifiedOnCorrectThreads$1(ListenableActionFutureTests.java:136)
```

--- a/server/src/test/java/org/elasticsearch/action/support/ListenableActionFutureTests.java
+++ b/server/src/test/java/org/elasticsearch/action/support/ListenableActionFutureTests.java
@@ -68,7 +68,6 @@ public class ListenableActionFutureTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/68772")
     public void testListenersNotifiedOnCorrectThreads() throws InterruptedException {
 
         final int adderThreads = between(1, 5);
@@ -90,7 +89,10 @@ public class ListenableActionFutureTests extends ESTestCase {
                     awaitSafe(barrier);
 
                     final AtomicBoolean isComplete = new AtomicBoolean();
-                    if (postComplete.get()) {
+                    if (completerThreads == 1 && postComplete.get()) {
+                        // If there are multiple completer threads then onResponse might return on one thread, and hence postComplete is
+                        // set, before the other completer thread notifies all the listeners. OTOH with one completer thread we know that
+                        // postComplete indicates that the listeners were already notified.
                         future.addListener(new ActionListener<>() {
                             @Override
                             public void onResponse(Void response) {

